ngraph.mk

```diff
@@ -0,0 +1,175 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# Including this .mk file has the following effects:
+#
+# - If any problem is noticed, calls $(error ...).
+#
+# - Assuming no error is noticed:
+#   - Sets the following variables (perhaps to the empty string):
+#     NGRAPH_CFLAGS  - Compiler args needed to build the MXnet-nGraph bridge code.
+#     NGRAPH_LDFLAGS - Linker args needed to build the MXnet-nGraph bridge code.
+#
+#   - In whatever way is appropriate, defines new goals and adds them as
+#     dependencies of the 'all' and/or 'clean' targets.
+#
+# Input variables:
+#   USE_NGRAPH
+#   USE_NGRAPH_DISTRIBUTED
+#   USE_NGRAPH_IE
+#   ADD_NGRAPH_LIBDIR_TO_MXNET_RPATH
+#   NGRAPH_DIR
+#   NGRAPH_EXTRA_CMAKE_FLAGS
+#   NGRAPH_EXTRA_MAKE_FLAGS
+
+#===================================================================================================
+# Check for some configuration problems...
+#===================================================================================================
+
+ifeq ($(USE_NGRAPH), 1)
+  ifeq ($(USE_MKLDNN), 1)
+    $(error "Cannot have both USE_NGRAPH=1 and USE_MKLDNN=1: they require different MKLDNN versions.")
+  endif
+endif
+
+#===================================================================================================
+
+ifeq ($(USE_NGRAPH), 1)
+  ifeq ($(NGRAPH_DIR),)
+    NGRAPH_INSTALL_DIR=$(ROOTDIR)/3rdparty/ngraph.install
+  else
+    NGRAPH_INSTALL_DIR=$(NGRAPH_DIR)
+  endif
+
+  NGRAPH_LIB_DIR=$(NGRAPH_INSTALL_DIR)/lib
+  NGRAPH_INCLUDE_DIR=$(NGRAPH_INSTALL_DIR)/include
+
+.PHONY: ngraph_clean
+
+  ifeq ($(NGRAPH_DIR),)
```

Comment:
The tabbing on this file is a little confusing, I guess that's due to Make's tab requirements?

Reply:
Unfortunately yeah.  For any rules, the first line must have no indentation, and the recipe lines must be intended by (at least) one leading tab character.

The other statements in a Makefile, however, are free to use indentation however they want (AFAIK).

Because I had several layers of `if` nesting, I thought the lesser of two evils was to indent nested `if`s.